core/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/LongTerms.java

```diff
@@ -91,6 +91,9 @@ public Bucket(long term, long docCount, InternalAggregations aggregations, boole
 
         @Override
         public String getKeyAsString() {
+            if (formatter instanceof ValueFormatter.BooleanFormatter) {
+                return term == 0 ? "false" : "true";
+            }
             return String.valueOf(term);
```

Comment:
I'm not too concerned as if you had a date field, you would rather run a date_histogram. @colings86 what do you think?